test/patterns/array_as_argument/ArrayAsArgument.java

```diff
@@ -0,0 +1,12 @@
+class ArrayAsArgument {
+    void noArgument() { // not here
+    }
+    void primitiveAsArgument(int x) { // not here
+    }
+    void arrayAsArgument(int[] x) { // here
+    }
+    void objectAsArgument(Object x) { // not here
+    }
```

Comment:
add the case when we have a generic array
```
    public GenSet(Class<E> c, int s) {
        // Use Array native method to create array
        // of a type only known at run time
        @SuppressWarnings("unchecked")
        final E[] a = (E[]) Array.newInstance(c, s);
        this.a = a;
    }
```

Reply:
@lyriccoder I think that the code in the example should be another pattern, because we are not using array as argument as this pattern implies